docs/agents/magpie.rst

```diff
@@ -0,0 +1,383 @@
+
+.. image:: ../_static/images/lyrebird_cmaps.png
+  :width: 1200
+  :alt: Lyrebird Colormaps
+
+There are additional colormaps ``white_cmap_fs`` and ``rainbow_cmap_fs`` that
+are identical to their counterparts above but instead map onto the range [-1, 1].
+The colormap ``phase_cmap`` is identical to ``rainbow_cmap`` but maps the range
+:math:`[0, \pi]`.
+
+
+If an equation's color is set to be dynamic, each equation value will be mapped
+to the range [0, 1] by interpolating between the min and max value in the
+equation's data buffer. This allows you to view changes equations that span a
+large range with a high dynamic-range, but makes direct channel-to-channel
+comparison impossible.
+
+Detector Layouts
+-----------------
+
+Grid Layout
+````````````
+The grid layout is a grid with 4096 elements (maximum number of channels a
+single smurf slot can stream)
+
+.. image:: ../_static/images/lyrebird_grid.png
+  :width: 400
+  :alt: Alternative text
+
+This layout contains 8 rows containing 512 detectors each, with the bottom
+row being band 0, and the top row band 7. This is the easiest layout to set up
+and is useful for viewing detector response as a function of their resonator
+frequency or band / channel id.
+
+Wafer Layout
+`````````````
+
+The Wafer layout takes in a det-map CSV file generated from the simonsobs 
+`detmap <https://github.com/simonsobs/DetMap.git>`_ package, and uses it
+to generate a focal-plane layout.
+
+.. image:: ../_static/images/lyrebird_wafer.png
+  :width: 400
+  :alt: Alternative text
+
+Additional Streaming Modes
+----------------------------
+
+Reading from G3Files
+``````````````````````
+
+It is  possible to tell magpie to stream data from existing G3Files instead of
+directly from the smurf-stream. To do this, simply set the ``src`` argument to
+be the filepath of the file you wish to stream. If you want to stream from
+multiple files in series, you can do this by pu::
+
+      {'agent-class': 'MagpieAgent',
+       'instance-id': 'magpie-crate1slot2',
+       'arguments': [
+         '--stream-id', 'crate1-slot2',
+
+         '--src', '/path/to/file1.g3', '/path/to/file2.g3',
+
+         '--dest', 8675,
+         '--delay', 5,
+         '--target-rate', 20
+         '--layout', 'wafer',
+
+         # Detmap CSV file
+         '--csv-file', '/home/jlashner/lyrebird_demo/detmap.csv',
+         '--offset', 0, 0,
+       ]},
+
+Streaming Fake Data
+``````````````````````
+
+To stream fake data, add the ``--fake-data`` argument. In this case
+you don't need to provide a data-source::
+
+      {'agent-class': 'MagpieAgent',
+       'instance-id': 'magpie-crate1slot2',
+       'arguments': [
+         '--stream-id', 'crate1-slot2',
+
+         '--dest', 8675,
+         '--delay', 5,
+         '--target-rate', 20
+         '--layout', 'wafer',
+
+         # Detmap CSV file
+         '--csv-file', '/home/jlashner/lyrebird_demo/detmap.csv',
+         '--offset', 0, 0,
+         '--fake-data',
+       ]},
+
```

Comment:
I would consider moving these under the "Configuration File Examples/OCS Site Config" section. (And possibly just describe the differences in the options, though I can understand wanting a whole block to copy.)